src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(parseDriversWithResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't find drivers in DB by car id "
+                    + carId, throwable);
+        }
+    }
+
+    private void insertDriversRelations(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement addDriversToCarStatement
+                         = connection.prepareStatement(query)) {
+            addDriversToCarStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                addDriversToCarStatement.setLong(2, driver.getId());
+                addDriversToCarStatement.executeUpdate();
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't insert drivers to car: " + car, throwable);
+        }
+    }
+
+    private Car parseCarWithManufacturer(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getObject("id", Long.class);
+        Manufacturer manufacturer = new Manufacturer(resultSet.getString("name"),
+                resultSet.getString("country"));
+        manufacturer.setId(manufacturerId);
+        Long carId = resultSet.getObject("car_id", Long.class);
+        String carModel = resultSet.getString("model");
+        Car car = new Car();
+        car.setModel(carModel);
+        car.setId(carId);
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private Driver parseDriversWithResultSet(ResultSet resultSet) throws SQLException {
+        Long newId = resultSet.getObject("id", Long.class);
+        String name = resultSet.getString("name");
+        String licenseNumber = resultSet.getString("license_number");
+        Driver driver = new Driver(name, licenseNumber);
+        driver.setId(newId);
+        return driver;
+    }
+
+    private void removeAllDriversRelations(Car car) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateCarStatement
+                         = connection.prepareStatement(query)) {
+            updateCarStatement.setLong(1, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't remove drivers relations "
+                    + car, throwable);
+        }
+    }
+
+    private void setDriversToListOfDrivers(List<Car> cars) {
+        if (cars.size() != 0) {
```

Comment:
this check is redundant, just run forEach() on the list, if it's empty you will skip it. 
And this logic is to small for a separate method